dash-user-guide-components/src/lib/components/PageMenu.react.js

```diff
@@ -0,0 +1,86 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {replace} from 'ramda';
+import { History } from '@plotly/dash-component-plugins';
+
+
+class PageMenu extends Component {
+    constructor(props) {
+        super(props);
+        this.onLocationChanged = this.onLocationChanged.bind(this);
+    }
+
+    onLocationChanged() {
+        this.forceUpdate();
+    }
+
+    componentDidMount() {
+        this._clearOnLocationChanged = History.onChange(this.onLocationChanged);
+    }
+
+    componentWillUnmount() {
+        this._clearOnLocationChanged();
+    }
+
+    render() {
+        /*
+         * Display links directly via setInterval because we don't know when the
+         * headers will be rendered in the DOM
+         */
+        function renderPageMenuLinks() {
+            const parent = document.getElementById('page-menu--links');
+            if(parent.innerText !== '' &&
+                // When the single page app location changes, update the page menu
+                parent.className === window.location.pathname
+            ) {
+                window.clearInterval(renderPageMenuLinks);
+                return;
+            }
+            const elements = document.querySelectorAll('h1, h2, h3, h4, h5, h6');
+            const links = [];
+            elements.forEach(el => {
+
+                if (!el.href) {
```

Comment:
```suggestion
                if (!el.id) {
```
or am I missing something?

Reply:
fixed 👍